picard/file_dilaog.py

```diff
@@ -0,0 +1,274 @@
+        cw=int(.7*self.l1w)
+        cw1=int(.143*self.l1w)
+        cw2=int(.143*self.l1w)
+        self.table.setColumnWidth(0,cw)
+        self.table.setColumnWidth(1,cw1)
+        self.table.setColumnWidth(2,cw2)
+        self.show_files(False)
+        self.table.resize(self.l1w,he-.26*he)
+        self.table.move(self.lw,50)
+        self.table.setShowGrid(False)
+        self.table.doubleClicked.connect(self.default_setting)
+
+        self.checkBox = QCheckBox("Show hidden files/folders",self)
+        self.checkBox.resize(200,30)
+        self.checkBox.move(wi-.365*wi,he-.17*he)
+        self.checkBox.toggled.connect(lambda:self.show_hidden(self.checkBox))
+
+        self.cb = QComboBox(self)
+        self.cb.addItem("All Supported Formats")
+        self.cb.addItems(self.formats_show)
+        self.cb.resize(200,30)
+        self.cb.move(wi-.183*wi,he-.17*he)
+        self.cb.currentIndexChanged.connect(self.format_modifier)
+
+        self.resize(wi,he)
+        sys.exit(self.exec_())
+
+    def format_modifier(self,i):
+
+        self.formats.clear()
+        self.formats_lower.clear()
+        if not self.cb.itemText(i) == "All Supported Formats":
+            self.formats = [self.cb.itemText(i)]
+        else:
+            self.formats=["MP3","FLAC","OGG","M4A","WMA","WAV","MP4","AAC","flv"]
+
+        for i in self.formats:self.formats_lower.append(i.lower())
+
+        self.show_files(False)
+
+    def default_setting(self):
+        indexes = self.table.selectionModel().selectedRows()
+        if len(indexes) == 1:
+            #for index in (indexes):
+            if os.path.isdir(os.path.join(self.path,self.table.item(indexes[0].row(),0).text()[6:])):
+                self.open_action(indexes)
+            else:
+                self.select_action(indexes)
+        else :
+            self.select_action(indexes)
+
+    def open_action(self,indexes):
+        if indexes == True:
+            indexes = self.table.selectionModel().selectedRows()
+        if len(indexes) == 1:
+            if os.path.isdir(os.path.join(self.path,self.table.item(indexes[0].row(),0).text()[6:])):
+                hola = (self.table.item(indexes[0].row(),0)).text()[6:]
+                self.path = os.path.join(self.path,hola)
+                self.show_files(False)
+
+    def select_action(self,indexes):
+        if indexes == True:
+            indexes = self.table.selectionModel().selectedRows()
+        self.send_things = []
+        for index in (indexes):
+            hola = (self.table.item(index.row(),0)).text()[6:]
+            if os.path.isdir(os.path.join(self.path,hola)):
+                self.recu_folder(self.send_things,os.path.join(self.path,hola))
+            else:
+                self.send_things.append(os.path.join(self.path,hola))
+        for  i in self.send_things:
+            print(i)
+        #print(self.send_things)
+        sys.exit()
+
+    def recu_folder(self,lis,path):
```

Comment:
please make sure you method names and variables names are appropriate. use something like `add_folder_recursive`